src/IIIF/IIIF/Serialisation/Deserialisation/CollectionItemConverter.cs

```diff
@@ -11,12 +12,25 @@ public class CollectionItemConverter : ReadOnlyConverter<ICollectionItem>
     {
         var jsonObject = JObject.Load(reader);
 
-        ICollectionItem collectionItem = jsonObject["type"].Value<string>() switch
+        var type = jsonObject["type"].Value<string>();
+        ICollectionItem collectionItem = null;
+            
+        // Look for consumer-provided mapping
+        if (serializer.Context.Context is IDictionary<string, Func<JObject, ICollectionItem>> customMappings

```

Comment:
note that this might cause issues if in one call you are trying to deserialize e.g. object that contains `List<ICollectionItem>` and e.g. `List<IUnrelatedInterface>`. You will either have to pass `IDictionary<string, Func<JObject, ICollectionItem>>` or ``IDictionary<string, Func<JObject, IUnrelatedInterface>>`, but not both at the same time.

How about sth like `IDictionary<Type, IDictionary<string, Func<JObject, object>>>`? I mean, it looks like eldritch nightmare, but:
```
if(serializer.COntext.Context is IDictionary<Type, IDictionary<string, Func<JObject, object>>> ctx){
  if(ctx.TryGetValue(typeof(ICollectionItem), out var customMappings){
    if(customMappings.TryGetValue(type, out var customMapping)){
      collectionItem = (ICollectionItem) customMapping(jsonObject);
    }
  }
}
```
that way each converter during one deserialization can look for its own set of supported mappings.

There is an argument of essentially wrapping this as sth akin to
```
public class IIIFSerializationContext {
// essentially IDictionary<Type, IDictionary<string, Func<JObject, object>>>, but maybe wrapped in like `<T>` methods to do the casting for the user?
}
```